src/spells.cpp

```diff
@@ -977,12 +977,12 @@ bool InstantSpell::canThrowSpell(const Creature* creature, const Creature* targe
 	return true;
 }
 
-bool InstantSpell::castSpell(Creature* creature)
+bool InstantSpell::castSpell(std::shared_ptr<Creature> creature)
 {
 	LuaVariant var;
 
 	if (casterTargetOrDirection) {
-		Creature* target = creature->getAttackedCreature();
+		std::shared_ptr<Creature> target = creature->getAttackedCreature();
```

Comment:
Could this be `const auto`
```suggestion
		const auto target = creature->getAttackedCreature();
```